src/ext/dispatch.c

```diff
@@ -60,18 +45,9 @@ static ddtrace_dispatch_t *lookup_dispatch(const HashTable *lookup, ddtrace_look
 }
 #else
 static ddtrace_dispatch_t *lookup_dispatch(const HashTable *lookup, ddtrace_lookup_data_t *lookup_data) {
-    zend_string *to_free = NULL, *key = lookup_data->function_name;
-
-    if (!ddtrace_is_all_lower(key)) {
-        key = zend_string_tolower(key);
-        to_free = key;
-    }
-
+    zend_string *key = zend_string_tolower(lookup_data->function_name);
```

Comment:
Followed the code of `zend_string_tolower_ex` Looks like the case of lowercased bas string is already handled well there. Great find!